tasks/go.py

```diff
@@ -366,3 +371,11 @@ def reset(ctx):
     # remove vendor folder
     print("Remove vendor folder")
     ctx.run("rm -rf ./vendor")
+
+@task
+def generate(ctx):
```

Comment:
Nit for later: the name feels a bit too generic for what this function does: for instance, it's difficult to understand without going into the specifics of the function that the:
```py
def build(...):
  ...
  # Generate go source from template
  generate(ctx)
  ...
```
line in `tasks/agent.py` means we apply `go generate` on `./pkg/status`.

Moreover, we use `go generate` in some other places (such as `tasks/cluster_agent.py`: https://github.com/DataDog/datadog-agent/blob/527af5ec063b7a8cef4f0227799f30861cbd4d37/tasks/cluster_agent.py#L66-L67, `tasks/dogstatsd.py`: https://github.com/DataDog/datadog-agent/blob/527af5ec063b7a8cef4f0227799f30861cbd4d37/tasks/dogstatsd.py#L95-L96, and `tasks/trace_agent.py`: https://github.com/DataDog/datadog-agent/blob/527af5ec063b7a8cef4f0227799f30861cbd4d37/tasks/trace_agent.py#L75-L76), so we could use this function in these cases too (with some tweaks to take different targets and options).


I'd either:
- Not change anything, but show explicitly that this function renders only what's specified in `GO_GENERATE_TARGETS` (eg. by adding a comment around its invocation in `tasks/agent.py` and `tasks/cluster_agent.py`) so that someone reading the build tasks understands what's happening without having to read the internal function's behavior),
- Make this function more generic by accepting a list of targets (and maybe a list of options if needed)

Reply:
I agree that it will make sense at some point to write a full-featured wrapper for `go generate`. In-between I took option #1 and added more meaningful comments.